diffrax/_integrate.py

```diff
@@ -73,23 +73,30 @@ class State(eqx.Module):
     dense_save_index: Optional[IntScalarLike]
 
 
-def _is_none(x):
+def _is_none(x: Any) -> bool:
     return x is None
 
 
-def _term_compatible(terms, term_structure):
+def _term_compatible(terms: PyTree[AbstractTerm], term_structure: PyTree) -> bool:
     def _check(term_cls, term):
         if get_origin(term_cls) is MultiTerm:
             if isinstance(term, MultiTerm):
                 [_tmp] = get_args(term_cls)
                 assert get_origin(_tmp) in (tuple, Tuple), "Malformed term_structure"
-                if not _term_compatible(term.terms, get_args(_tmp)):
-                    raise ValueError
-            else:
-                raise ValueError
+                if _term_compatible(term.terms, get_args(_tmp)):
+                    return
         else:
-            if not isinstance(term, term_cls):
-                raise ValueError
+            term_args_expected = get_args(term_cls)
+            control_compatible = True
+            if term_args_expected:
```

Comment:
Here's another safe-programming-style hobby-horse of mine :)
Always explicitly use booleans in `if` statements. IIUC, I think we're (currently) expecting this to be either a `None` or a length-1 tuple of the control type? In that case I would write this as:
```python
if term_args_expected is None:
    control_compatible = True
else:
    (term_arg_expected,) = term_args_expected
    ...
```